crates/ty_server/src/server/api.rs

```diff
@@ -135,7 +135,52 @@ where
     }))
 }
 
-fn background_request_task<R: traits::BackgroundDocumentRequestHandler>(
+#[expect(dead_code)]
```

Comment:
Could you add a comment saying why this code is kept around even though it isn't used?

Reply:
I split the workspace diagnostics work into two PRs to ease up the review process. This is going to be used in the follow-up PR (#18939). Sorry, I should've highlighted this as a review comment!